apps/nextjs/src/components/AppComponents/Chat/chat-lessonPlanDisplay.tsx

```diff
@@ -158,28 +204,19 @@ export const LessonPlanDisplay = ({
       )}
 
       <div className="flex w-full flex-col justify-center">
-        {organiseSections.map((section) => {
-          const trigger = lessonPlan[section.trigger];
-          return (
-            !!trigger &&
-            section.dependants.map((dependant) => {
-              const value = lessonPlan[dependant];
-              if (value !== null && value !== undefined) {
-                return (
-                  <DropDownSection
-                    key={dependant}
-                    objectKey={dependant}
-                    sectionRefs={sectionRefs}
-                    value={value}
-                    userHasCancelledAutoScroll={userHasCancelledAutoScroll}
-                    documentContainerRef={documentContainerRef}
-                    showLessonMobile={showLessonMobile}
-                  />
-                );
-              }
-            })
-          );
-        })}
+        {allSectionsInOrder.map((section) => (
```

Comment:
Rather than selectively rendering each section, we render them all with a visible attribute. This reduces thrash with the DIV refs so that when we try to scroll to each section during streaming, the DIV ref is not null. This comment is the result of several hours of work!